libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -296,39 +297,17 @@ public void cleanResyncGhosts(long syncId) throws JSONException {
             throw new SmartSyncException("Cannot run cleanResyncGhosts:" + syncId + ": wrong type:" + sync.getType());
         }
         final String soupName = sync.getSoupName();
-        final String idFieldName = sync.getTarget().getIdFieldName();
         final SyncDownTarget target = (SyncDownTarget) sync.getTarget();
 
-        /*
-         * Fetches list of IDs present in local soup that have not been modified locally.
-         */
-        final Set<String> localIds = target.getNonDirtyRecordIds(this, soupName, idFieldName);
-
-        /*
-         * Fetches list of IDs still present on the server from the list of local IDs
-         * and removes the list of IDs that are still present on the server.
-         */
-        final Set<String> remoteIds = target.getListOfRemoteIds(this, localIds);
-        if (remoteIds != null) {
-            localIds.removeAll(remoteIds);
-        }
-
-        // Deletes extra IDs from SmartStore.
-        int localIdSize = localIds.size();
+        // Ask target to clean up ghosts
```

Comment:
Delegating to the target (required for advanced targets like ParentChildrenSyncDownTarget).